core/spec/models/spree/promotion_code_batch_spec.rb

```diff
@@ -0,0 +1,71 @@
+require "spec_helper"
+
+describe Spree::PromotionCodeBatch, type: :model do
+  let(:promotion) { create(:promotion) }
+  let(:base_code) { "abc" }
+  let(:number_of_codes) { 1000 }
+  let(:error) { nil }
+  subject do
+    described_class.create!(
+      promotion_id: promotion.id,
+      base_code: base_code,
+      number_of_codes: number_of_codes,
+      error: error,
+      email: "test@email.com"
+    )
+  end
+
+  describe "#status" do
+    context "with error present" do
+      let(:error) { "hello" }
+
+      it "returns the error" do
+        expect(subject.status).to eq error
+      end
+    end
+    context "with number_of_codes == promotion_codes.count" do
+      before do
+        allow(subject).to receive_message_chain("promotion_codes.count")
+          .and_return(number_of_codes)
+      end
+
+      it "returns 'All codes created'" do
+        expect(subject.status).to eq "All codes created"
+      end
+    end
+
+    context "with codes still processing" do
+      before do
+        allow(subject).to receive_message_chain("promotion_codes.count")
+          .and_return(number_of_codes - 1)
+      end
+
+      it "returns the number of codes processed" do
+        expect(subject.status)
+          .to eq("Processing: #{number_of_codes - 1} / #{number_of_codes}")
+      end
+    end
+  end
+
+  describe "#process" do
+    it "should call the worker" do
+      allow(Spree::PromotionCodeBatchJob).to receive(:perform_later)
+
+      subject.process
+
+      expect(Spree::PromotionCodeBatchJob)
+        .to have_received(:perform_later)
+        .with(subject)
+    end
+
+    context "with a batch that already started" do
+      before do
+        allow(subject).to receive_message_chain("promotion_codes.count")
+          .and_return(1)
```

Comment:
can you kill the stubbing in this? I don't think its too costly to create a promo batch with a code to avoid the message chain